### PR TITLE
Assignment add view update grade

### PR DESCRIPTION
**Pull request by imposyndro**
Here are the files for adding, viewing, updating and grades for assignments. Since the previous pull request has not yet been approved, the changes from the previous pull request are included to ensure cohesion.

### CODE REVIEW

File: src/components/instructor/AssignmentsView.js [hunk 1]
```diff
@@ -9,16 +11,83 @@ import {useLocation} from 'react-router-dom'
 // display a table with columns 
 // assignment id, title, dueDate and buttons to grade, edit, delete each assignment
 
-const AssignmentsView = (props) => {
```
Comment (abourdeaux-csumb) [inline src/components/instructor/AssignmentsView.js:12]: This shouldn't be deleted

File: src/components/instructor/AssignmentsView.js [hunk 2]
```diff
@@ -9,16 +11,83 @@ import {useLocation} from 'react-router-dom'
...
+const [assignments, setAssignments] = useState([]);
+const [message, setMessage] = useState('');
 
-    const location = useLocation();
-    const {secNo, courseId, secId} = location.state;
```
Comment (abourdeaux-csumb) [inline src/components/instructor/AssignmentsView.js:15]: I don't think these should be deleted either.

File: src/components/instructor/AssignmentsView.js [hunk 3]
```diff
@@ -9,16 +11,83 @@ import {useLocation} from 'react-router-dom'
...
-    );
-}
+useEffect(() => {
+    fetchAssignments();
+}, []);
 
-export default AssignmentsView;
+const fetchAssignments = async () => {
+    try {
+        const response = await fetch(`${SERVER_URL}/sections/${secNo}/assignments`);
```
Comment (abourdeaux-csumb) [inline src/components/instructor/AssignmentsView.js]: Where does secNo come from?

File: src/components/instructor/AssignmentsView.js [hunk 4]
```diff
@@ -9,16 +11,83 @@ import {useLocation} from 'react-router-dom'
...
+        if (response.ok) {
+            const assignments = await response.json();
```
Comment (abourdeaux-csumb) [inline src/components/instructor/AssignmentsView.js]: I would change the variable here to `data` or something similar. The name `assignments` is already taken in the outside scope, I would guess that this would cause issues.

File: src/components/instructor/AssignmentsView.js [hunk 5]
```diff
@@ -9,16 +11,83 @@ import {useLocation} from 'react-router-dom'
...
+                            </Button>
+                            <Button variant="contained" color="error" onClick={() => deleteAssignment(assignment.assignmentId)}>
+                                Delete
+                            </Button>
+                        </TableCell>
+                    </TableRow>
+                ))}
+            </TableBody>
+        </Table>
+        <AssignmentAdd save={() => fetchAssignments()} />
```
Comment (abourdeaux-csumb) [inline src/components/instructor/AssignmentsView.js]: Does this work?

Comment (abourdeaux-csumb) [reply src/components/instructor/AssignmentsView.js]: For adding a new assignment, I mean.

File: src/components/instructor/AssignmentsView.js [hunk 6]
```diff
@@ -9,16 +11,83 @@ import {useLocation} from 'react-router-dom'
...
+                </TableRow>
+            </TableHead>
+            <TableBody>
+                {assignments.map((assignment) => (
+                    <TableRow key={assignment.assignmentId}>
+                        <TableCell>{assignment.assignmentId}</TableCell>
+                        <TableCell>{assignment.title}</TableCell>
+                        <TableCell>{assignment.dueDate}</TableCell>
+                        <TableCell>
+                            <Button variant="contained" color="primary" onClick={() => {/* navigate to grade page */}}>
```
Comment (abourdeaux-csumb) [inline src/components/instructor/AssignmentsView.js]: This button doesn't do anything right now.

File: src/components/instructor/AssignmentsView.js [hunk 7]
```diff
@@ -9,16 +11,83 @@ import {useLocation} from 'react-router-dom'
...
+                                Grade
+                            </Button>
+                            <Button variant="contained" color="secondary" onClick={() => {/* navigate to edit page */}}>
```
Comment (abourdeaux-csumb) [inline src/components/instructor/AssignmentsView.js]: Neither does this one.

File: src/components/instructor/AssignmentUpdate.js [hunk 8]
```diff
@@ -1,17 +1,93 @@
...
+            const response = await fetch(`${SERVER_URL}/assignments`, {
+                method: 'PUT',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(updatedAssignment),
+            });
+            if (response.ok) {
+                setMessage('Assignment updated successfully');
+                save(updatedAssignment); // Assuming save will update the parent component's state
```
Comment (abourdeaux-csumb) [inline src/components/instructor/AssignmentUpdate.js]: What does this mean 'assuming'? This should be tested, so assumptions shouldn't really come into play here,

File: src/components/student/CourseEnroll.js [hunk 9]
```diff
@@ -1,5 +1,7 @@
 import React, {useState, useEffect} from 'react';
```
Comment (abourdeaux-csumb) [inline src/components/student/CourseEnroll.js]: Including my changes in this PR will probably create a merge conflict. I would revert the changes I made.

File: src/components/instructor/AssignmentGrade.js [hunk 10]
```diff
@@ -1,21 +1,100 @@
...
+
+    useEffect(() => {
+        fetchGrades();
+    }, []);
+
+    const fetchGrades = async () => {
+        try {
+            const response = await fetch(`${SERVER_URL}/assignment/${props.assignmentId}/grades`);
+            if (response.ok) {
+                const grades = await response.json();
```
Comment (abourdeaux-csumb) [inline src/components/instructor/AssignmentGrade.js]: Would change `grades` to `data` considering `grades` is already defined above

File: src/components/instructor/AssignmentAdd.js [hunk 11]
```diff
@@ -1,19 +1,92 @@
...
+            const response = await fetch(`${SERVER_URL}/assignments`, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(assignment),
+            });
+            if (response.ok) {
+                setMessage('Assignment added successfully');
+                props.save(assignment); // Assuming props.save will update the parent component's state
```
Comment (abourdeaux-csumb) [inline src/components/instructor/AssignmentAdd.js]: Again, shouldn't be assuming things.

**Review by abourdeaux-csumb — CHANGES_REQUESTED** [review]
A number of issues left in the comments. I would revert the changes that were in my PR as they may cause conflicts when attempting to merge.

File: src/components/instructor/AssignmentsView.js (hunk 3 above)
Comment (sarp8099) [reply src/components/instructor/AssignmentsView.js]: the header const array from what I can tell

File: src/components/instructor/AssignmentsView.js (hunk 7 above)
Comment (sarp8099) [reply src/components/instructor/AssignmentsView.js]: I'd say leaving it for now should be fine if we plan to implement functionality later.

File: src/components/instructor/AssignmentsView.js (hunk 6 above)
Comment (sarp8099) [reply src/components/instructor/AssignmentsView.js]: I'd say leaving it for now should be fine if we plan to implement functionality later.

File: src/components/instructor/AssignmentsView.js (hunk 5 above)
Comment (sarp8099) [reply src/components/instructor/AssignmentsView.js]: I think this is standard practice

File: src/components/instructor/AssignmentsView.js (hunk 1 above)
Comment (sarp8099) [reply src/components/instructor/AssignmentsView.js:12]: This is required for the functionality of the file

File: src/components/instructor/AssignmentUpdate.js (hunk 8 above)
Comment (sarp8099) [reply src/components/instructor/AssignmentUpdate.js]: I believe that the handleSave function should be used instead because you already defined it in assignmentUpdate.js. Props in this case, would be an assignment object and with the implementation of the other files, you should be able to see what functions those objects can call.

File: src/components/instructor/AssignmentAdd.js (hunk 11 above)
Comment (sarp8099) [reply src/components/instructor/AssignmentAdd.js]: I believe that the handleSave function should be used instead because you already defined it in assignmentUpdate.js. Props in this case, would be an assignment object and with the implementation of the other files, you should be able to see what functions those objects can call.

**Review by sarp8099 — CHANGES_REQUESTED** [review]
The code seems a bit off with how items are logically processed. Make sure to compare to the admin folder for some reference if you are not completely comfy with react.

**Comment by abourdeaux-csumb** [comment]
Leaving the remaining code review for @sarp8099 as communicated over discord.
